Updates for Mix

srtfisher opened the pull request:
Add parts of #117 to improve the Mix integration with Mantle. Sets up for automatic entries and block enqueuing.

anubisthejackle commented on package.json:
```diff
@@ -1,21 +1,122 @@
+  "devDependencies": {
+    "@types/jest": "^29.0.3",
+    "@babel/preset-react": "^7.18.6",
+    "@tinypixelco/laravel-mix-wp-blocks": "^1.2.0",
```
This should handle all of the @wordpress packages for us, no?

srtfisher replied:
What I struggle with is IDE completion. Without these packages, I can't get it unless I use a `@types/` or something. Will dig in.

stevenslack commented on jsconfig.json:
```diff
@@ -0,0 +1,8 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@/*": ["*"]
+    }
+  }
+}
```
If Typescript is supported this file could be eliminated in favor of the `tsconfig.json`

stevenslack commented on package.json:
```diff
@@ -1,21 +1,122 @@
+    "test:watch": "jest --watch",
+    "pretest": "check-node-version --package",
+    "test": "jest",
+    "preupdate-dependencies": "check-node-version --package",
+    "update-dependencies": "node bin/update-dependencies.js",
```
If the update dependencies script is swapped in favor or `@wordpress/scripts` you could add the script here. Example in the starter plugin: https://github.com/alleyinteractive/wp-starter-plugin/blob/feature/wp-scripts/package.json#L33-L34

stevenslack commented on src/webpack/discover-entries.js:
```diff
@@ -0,0 +1,70 @@
+      .reduce((acc, item) => {
+        const entry = item
+          .replace(`${rootPath}/${this.entryPath}/`, '')
+          .replace(/\/index\.jsx?/, '');
```
Maybe supply a regex for `.tsx` as well?

stevenslack commented on webpack.mix.js:
```diff
@@ -10,20 +14,90 @@ require('laravel-mix-polyfill');
+|
+| Discover index.ts files within a specific directory using Typescript:
+|
+|  mix.discover('entries', { type: 'ts' });
```
Could it not detect `ts` files as well as `js` files without specification? Looks like you have the logic here: https://github.com/alleyinteractive/mantle/pull/118/files#diff-0da3eaa1f1380aee7113ee4a1c26d26b1e425f1b03994413a7fdf5b068b654bfR49

I am not that familiar with Mix, so I could be missing something.

stevenslack commented on webpack.mix.js:
```diff
@@ -10,20 +14,90 @@ require('laravel-mix-polyfill');
+*/
+
+mix.discover('entries');
+mix.discover('blocks', { type: 'block' });
```
I would love to discuss compiling blocks with `block.json` involved!

stevenslack reviewed (approved):
Nothing blocking, but added some comments for consideration. 🐞

srtfisher commented:
@stevenslack thank you for all that feedback, that really helped. I've switched to use `.ts` files by default and included all the relevant packages. I also switched to using `@wordpress/scripts`, too. Really neat find!

I'd love to regroup on this and the starter plugins to solidify shared initiates -- this one is almost ready. @anubisthejackle is wrapping up Gutenberg block generation and then we'll write up some documentation on it that make it very easy to generate blocks with Mantle.
